File: src/test/java/ru/jerael/booktracker/backend/data/mapper/PublisherDataMapperTest.java
package ru.jerael.booktracker.backend.data.mapper;

import org.junit.jupiter.api.Test;
import ru.jerael.booktracker.backend.data.db.entity.PublisherEntity;
import ru.jerael.booktracker.backend.domain.model.publisher.Publisher;
import java.util.UUID;
import static org.junit.jupiter.api.Assertions.assertEquals;

class PublisherDataMapperTest {
    private final PublisherDataMapper publisherDataMapper = new PublisherDataMapper();

    private final UUID id = UUID.fromString("a95f2c31-acd2-4413-83ea-c775d6cda257");
    private final String name = "Publisher Name";

    @Test
    void toEntity() {
        Publisher publisher = new Publisher(id, name);

        PublisherEntity entity = publisherDataMapper.toEntity(publisher);

        assertEquals(id, entity.getId());
        assertEquals(name, entity.getName());
    }

    @Test
    void toDomain() {
        PublisherEntity entity = new PublisherEntity();
        entity.setId(id);
        entity.setName(name);

        Publisher publisher = publisherDataMapper.toDomain(entity);

        assertEquals(id, publisher.id());
        assertEquals(name, publisher.name());
    }
}

File: src/test/kotlin/ru/jerael/booktracker/backend/data/mapper/PublisherDataMapperTest.kt
package ru.jerael.booktracker.backend.data.mapper

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import ru.jerael.booktracker.backend.factory.publisher.PublisherDomainFactory
import ru.jerael.booktracker.backend.factory.publisher.PublisherEntityFactory

class PublisherDataMapperTest {
    private val mapper = PublisherDataMapper()
    
    @Test
    fun `toEntity should map Publisher to PublisherEntity`() {
        val publisher = PublisherDomainFactory.createPublisher()
        
        val entity = mapper.toEntity(publisher)
        
        with(entity) {
            assertEquals(publisher.id, id)
            assertEquals(publisher.name, name)
        }
    }
    
    @Test
    fun `toDomain should map PublisherEntity to Publisher`() {
        val entity = PublisherEntityFactory.createPublisherEntity()
        
        val publisher = mapper.toDomain(entity)
        
        with(publisher) {
            assertEquals(entity.id, id)
            assertEquals(entity.name, name)
        }
    }
}